airflow/sensors/base_sensor_operator.py

```diff
@@ -70,6 +72,7 @@ def __init__(self,
                  timeout: float = 60 * 60 * 24 * 7,
                  soft_fail: bool = False,
                  mode: str = 'poke',
+                 downstream_skip=None,
```

Comment:
Why is the default None? If you want the default to be the original behaviour before this PR, it should default to True and handled accordingly in `_timeout_task()`.

Reply:
@yuqian90 

This is because the default behavior is different when soft_fail is not set / set False or when soft_fail=True.
When setting `soft_fail = True` ,  All downstream tasks are skipped forcibly.
When setting `soft_fail = False`,  All downstream tasks are not skipped forcibly.

However, I will fix it according to your opinion.
>The original behaviour is to just raise AirflowSensorTimeout.

Reply:
Thinking about this, I find it unnecessary to provide the `downstream_skip` option. We should just remove the call to `_do_skip_downstream_tasks()` altogether. That can be the only behaviour of `soft_fail`. 

My reason is that whoever not finding `_do_skip_downstream_tasks()` counter-intuitive is most likely using only `all_success` trigger rule in their DAGs. If we remove `_do_skip_downstream_tasks()`, those DAGs will not be affected. Things will continue to work the same. That said, you should still highlight this in the CHANGELOG so people are aware.

Reply:
@yuqian90 
 I very much agree with you.
>We should just remove the call to _do_skip_downstream_tasks() altogether. 
Indeed, when all_success trigger_rule is set, the downstream task is automatically skipped when the upstream task is skipped.

I will fix it!